### PR TITLE
Editables

### PR DESCRIPTION
**Pull request by p2kmgcl**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/FragmentEntryLink.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+const resolveEditableValue = (fragmentEntryLink, editable, pageStatus) => {
+	const data =
+		fragmentEntryLink.editableValues[
+			'com.liferay.fragment.entry.processor.editable.EditableFragmentEntryProcessor'
+		][editable.getAttribute('id')];
+
+	let value = data;
+
+	// TODO use Andres' function
```
Comment (julien) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/FragmentEntryLink.js]: :rofl:

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js [hunk 2]
```diff
@@ -24,141 +24,15 @@ import {DispatchContext} from '../reducers/index';
...
-	[LAYOUT_DATA_ITEM_TYPES.column]: Column,
-	[LAYOUT_DATA_ITEM_TYPES.container]: Container,
-	[LAYOUT_DATA_ITEM_TYPES.fragment]: Fragment,
-	[LAYOUT_DATA_ITEM_TYPES.root]: Root,
-	[LAYOUT_DATA_ITEM_TYPES.row]: Row
+	[LAYOUT_DATA_ITEM_TYPES.column]: React.forwardRef(Column),
+	[LAYOUT_DATA_ITEM_TYPES.container]: React.forwardRef(Container),
+	[LAYOUT_DATA_ITEM_TYPES.fragment]: React.forwardRef(Fragment),
+	[LAYOUT_DATA_ITEM_TYPES.root]: React.forwardRef(Root),
+	[LAYOUT_DATA_ITEM_TYPES.row]: React.forwardRef(Row)
```
Comment (julien) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js]: Let's talk about this when you have time

**Comment by julien** [comment]
@p2kmgcl since this is WIP and we still need to work on this, I "approved" but we still need to get together to define next tasks and possibly finish the things that are in `TODO` comments. I'd also like to show this to @wincent when he has a bit of time. Thanks!

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js [hunk 3]
```diff
@@ -207,7 +81,45 @@ const LAYOUT_DATA_FLOATING_TOOLBAR_TYPES = {
...
+	);
+
+	const mainItem = layoutData.items[layoutData.rootItems.main];
+
+	const isMounted = useIsMounted();
+	const shouldUpdateLayoutData = useRef();
+
+	useEffect(() => {
+		if (isMounted()) {
+			// Only run this after first render
```
Comment (wincent) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js]: How is this different from just passing an empty dependencies array to `useEffect`? (I know you may need to suppress a lint to do that.)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js (hunk 2 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js]: We just talked about it, everything is fine, MOVE

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js (hunk 3 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js]: This is ugly, horrible, horrendous, terrible, catastrophic, we are removing it NOW (maybe soon)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/FragmentEntryLink.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	if (typeof value !== 'string') {
+		value = data.defaultValue;
+	}
+
+	// TODO manage mapped fields
+	// both in display pages and content pages.
+	// They should be fetched from backend dynamically
+	return new Promise(resolve => {
+		resolve(`${value}${pageStatus.languageId}`);
+	});
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/FragmentEntryLink.js]: image

[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/FragmentEntryLink.js]: :( that was a mindexplode gif

**Comment by p2kmgcl** [comment]
https://github.com/victorg1991/liferay-portal/pull/129

**Pull request closed without merging**